internal/app/wwctl/configure/nfs/main.go

```diff
@@ -56,35 +48,57 @@ func Configure(show bool) error {
 		fmt.Fprintf(fstab, "sysfs /sys sysfs defaults 0 0\n")
 		fmt.Fprintf(fstab, "proc /proc proc defaults 0 0\n")
 
-		for _, export := range controller.Nfs.Exports {
-			fmt.Fprintf(exports, "%s %s/%s(sync)\n", export, controller.Network, controller.Netmask)
-			fmt.Fprintf(fstab, "%s:%s %s nfs defaults 0 0\n", controller.Ipaddr, export, export)
-		}
-
-		fmt.Printf("Enabling and restarting the NFS services\n")
-		if controller.Nfs.SystemdName == "" {
-			err := util.SystemdStart("nfs-server")
+		if controller.Nfs.Enabled {
+			exports, err := os.OpenFile("/etc/exports", os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
 			if err != nil {
-				return errors.Wrap(err, "failed to start nfs-server")
+				wwlog.Printf(wwlog.ERROR, "%s\n", err)
+				os.Exit(1)
 			}
-		} else {
-			err := util.SystemdStart(controller.Nfs.SystemdName)
-			if err != nil {
-				return errors.Wrap(err, "failed to start")
+			defer exports.Close()
+
+			fmt.Fprintf(exports, "# This file was written by Warewulf (wwctl configure nfs)\n")
+
+			for _, export := range controller.Nfs.Exports {
+				fmt.Fprintf(exports, "%s %s/%s(sync)\n", export, controller.Network, controller.Netmask)
+				fmt.Fprintf(fstab, "%s:%s %s nfs defaults 0 0\n", controller.Ipaddr, export, export)
 			}
-		}
 
+			for _, export := range controller.Nfs.ExportsExtended {
+				fmt.Fprintf(exports, "%s %s/%s(%s)\n", export.Path, controller.Network, controller.Netmask, export.Options)
+				if export.Mount {
+					fmt.Fprintf(fstab, "%s:%s %s nfs defaults 0 0\n", controller.Ipaddr, export.Path, export.Path)
```

Comment:
I wonder if there would be useful to extend options to the client side mounts as well... Thoughts?

Reply:
I agree, but warewulf.conf should do the service stuff and nodes.conf should do the client stuff. I don't like mixing functions. I would remove the "mount" key here, since that configuration should reside under nodes.conf.

I'm noticing all of this seems NFS3. In 2021, shouldn't we be defaulting to NFSv4 standards.